### ISSUE TITLE
new publisher tries to publish already existing files

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#6252]
example in
```
belforte@crab-prod-tw02/sixie> pwd
/data/container/Publisher_schedd/logs/tasks/sixie
belforte@crab-prod-tw02/sixie> ls -ltr
total 1300
-rw-r--r--. 1 crab3 1000 1326150 Nov 13 12:29 201106_010251:sixie_crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2.log
belforte@crab-prod-tw02/sixie> 

```

**Comment by belforte** [comment]
error is
```
HTTP Error 400: Duplicated lfn name wh
en inserting                    files. Traceback (most recent call last):
  File "/build/cmsbld/jenkins/workspace/CRABServer_BuildOnRelease/w/BUILD/slc7_amd64_gcc630/cms/crab
taskworker/v3.201111/CRABServer-v3.201111/build/lib/Publisher/TaskPublish.py", line 788, in publishI
nDBS3
  File "/data/srv/TaskManager/v3.201111/slc7_amd64_gcc630/cms/dbs3-client/3.13.1-comp/lib/python2.7/
site-packages/dbs/apis/dbsClient.py", line 346, in insertBulkBlock
    result =  self.__callServer("bulkblocks", data=blockDump, callmethod='POST' )
  File "/data/srv/TaskManager/v3.201111/slc7_amd64_gcc630/cms/dbs3-client/3.13.1-comp/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 203, in __callServer
    self.__parseForException(http_error)
  File "/data/srv/TaskManager/v3.201111/slc7_amd64_gcc630/cms/dbs3-client/3.13.1-comp/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 230, in __parseForException
    raise HTTPError(http_error.url, data['exception'], data['message'], http_error.header, http_error.body)
HTTPError: HTTP Error 400: Duplicated lfn name when inserting                    files. 

```

which comes from:
```
2020-11-12 12:42:20,785:INFO: Published output config.
2020-11-12 12:42:20,785:INFO: About to insert dataset: {'dataset': u'/ttH_HToSS_SToBB_ms40_pl1000/si
xie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df761
6e8/USER', 'physics_group_name': 'CRAB3', 'dataset_access_type': 'VALID', 'last_modification_date': 
1605181340, 'processed_ds_name': u'sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSI
Moutput-8893946ae0a9fb35ec082af8df7616e8', 'data_tier_name': u'USER'}
2020-11-12 12:42:20,794:INFO: Found 49 files not already present in DBS which will be published.
2020-11-12 12:42:20,854:ERROR: Error for files: [u'/store/group/phys_exotica/privateProduction/GS/Ru
nIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_ST
oBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_814.root', ....
```
where the list of file names in the [...] apppears to have way more than 49 entries

**Comment by belforte** [comment]
The list of file names in comment above is simply wrong, I need to fix code there,
also the part which saves the failed block to disk is buggy.

I rerun with one file per block. It fails on this one:
```
{'acquisition_era': {'acquisition_era_name': 'CRAB', 'start_date': 0},
 'block': {'block_name': u'/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8/USER#b63a207b-3a0a-489d-91ee-f094c34c0b9c',
           'block_size': 103166254,
           'file_count': 1,
           'open_for_writing': 0,
           'origin_site_name': u'T2_US_Caltech'},
 'dataset': {'data_tier_name': u'USER',
             'dataset': u'/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8/USER',
             'dataset_access_type': 'VALID',
             'last_modification_date': 1605290307,
             'physics_group_name': 'CRAB3',
             'processed_ds_name': u'sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8'},
 'dataset_conf_list': [{'app_name': 'cmsRun',
                        'global_tag': 'crab3_tag',
                        'output_module_label': 'o',
                        'pset_hash': u'8893946ae0a9fb35ec082af8df7616e8',
                        'release_version': u'CMSSW_10_2_16_UL'}],
 'file_conf_list': [{'app_name': 'cmsRun',
                     'global_tag': 'crab3_tag',
                     'lfn': u'/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root',
                     'output_module_label': 'o',
                     'pset_hash': u'8893946ae0a9fb35ec082af8df7616e8',
                     'release_version': u'CMSSW_10_2_16_UL'}],
 'file_parent_list': [],
 'files': [{'adler32': u'6b4a79d1',
            'check_sum': u'1514223708',
            'event_count': 100,
            'file_lumi_list': [{'lumi_section_num': 528, 'run_num': 1}],
            'file_size': 103166254,
            'file_type': 'EDM',
            'logical_file_name': u'/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root'}],
 'primds': {'primary_ds_name': u'ttH_HToSS_SToBB_ms40_pl1000',
            'primary_ds_type': 'mc'},
 'processing_era': {'description': 'CRAB3_processing_era',
                    'processing_version': 1}}[crab3@crab-prod-tw02 Publisher]$ cat /tmp/failed-block-at-1605290762.56.txt;echo""
{'acquisition_era': {'acquisition_era_name': 'CRAB', 'start_date': 0},
 'block': {'block_name': u'/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8/USER#b63a207b-3a0a-489d-91ee-f094c34c0b9c',
           'block_size': 103166254,
           'file_count': 1,
           'open_for_writing': 0,
           'origin_site_name': u'T2_US_Caltech'},
 'dataset': {'data_tier_name': u'USER',
             'dataset': u'/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8/USER',
             'dataset_access_type': 'VALID',
             'last_modification_date': 1605290307,
             'physics_group_name': 'CRAB3',
             'processed_ds_name': u'sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8'},
 'dataset_conf_list': [{'app_name': 'cmsRun',
                        'global_tag': 'crab3_tag',
                        'output_module_label': 'o',
                        'pset_hash': u'8893946ae0a9fb35ec082af8df7616e8',
                        'release_version': u'CMSSW_10_2_16_UL'}],
 'file_conf_list': [{'app_name': 'cmsRun',
                     'global_tag': 'crab3_tag',
                     'lfn': u'/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root',
                     'output_module_label': 'o',
                     'pset_hash': u'8893946ae0a9fb35ec082af8df7616e8',
                     'release_version': u'CMSSW_10_2_16_UL'}],
 'file_parent_list': [],
 'files': [{'adler32': u'6b4a79d1',
            'check_sum': u'1514223708',
            'event_count': 100,
            'file_lumi_list': [{'lumi_section_num': 528, 'run_num': 1}],
            'file_size': 103166254,
            'file_type': 'EDM',
            'logical_file_name': u'/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root'}],
 'primds': {'primary_ds_name': u'ttH_HToSS_SToBB_ms40_pl1000',
            'primary_ds_type': 'mc'},
 'processing_era': {'description': 'CRAB3_processing_era',
                    'processing_version': 1}}

```

i.e. on file
```
/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root
```

**Comment by belforte** [comment]
indeed that file is already present in DBS/phys03, but in a different dataset
I asked the user (Si Xie) if this coudl have been some configuration error on his side
```
It looks like it tries to inset this file
/store/group/phys_exotica/privateProduction/GS/RunIIFall18/GENSIM/ttH_HToSS_SToBB_ms40_pl1000/v1/ttH_HToSS_SToBB_ms40_pl1000/crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2/201106_010251/0000/HIG-RunIIFall18wmLHEGS-ggH_CPfromJingyu_528.root
in dataset:
/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_RAWSIMoutput-8893946ae0a9fb35ec082af8df7616e8/USER
but it fails because that file is already in
/ttH_HToSS_SToBB_ms40_pl1000/sixie-crab_Fall18_ttH_HToSS_SToBB_ms40_pl1000_GENSIM_batch2_LHEoutput-8893946ae0a9fb35ec082af8df7616e8/USER

note that one dataset has LHEoutput, the other has RAWSIMoutput in same place

I am a bit puzzled becasue I would have expected that those names are created by CRAB in
such a way to avoid this. Is it possible that this was simply a misconfiguration at
your end ? 
```

**Comment by belforte** [comment]
That task is a mess, some files are marked in transfersdb with aso_worker=asoless, some with schedd, some none..
The task was submitted before the switch, does not have USE_NEW_PUBLISHER file in its SPOOL_DIR and some files ended up in new publisher only because aso_worker had not been filled by PostJon.

Well.. the transition was a bit hurried up, ideally we should have waited one month at every step to make sure system was flushed.. but that was not possible.

I communicated with the user (Si Xie) who is puzzled as well how a file *GENSIM* could have been inserted in ad *RAWSim* dataset. But since this was the only publication failures in first two days... I will try to ignore it and focus on improving reporting since now that I fixed #6254 scripts do not crash when publication fails and nothing is clearly reported.
